publish.js

```diff
@@ -0,0 +1,272 @@
+ */
+publish.getCLIOpts = function () {
+    var opts = {
+        options: {}
+    };
+    // print process.argv
+    process.argv.forEach(function(val, index) {
+        if (index > 1) {
+            var opt = val.split("=");
+
+            // convert "false" to boolean
+            if (opt[1] === "false") {
+                opt[1] = false;
+            }
+
+            opts[opt[0]] = opt.length < 2 ? true : opt[1];
+        }
+    });
+    return opts;
+};
+
+/**
+ * Creaes a number with 0's padded on the left
+ *
+ * @param num {Number}
+ * @param with {Number} - the min-width of the number,
+ *                        if the number is shorter it will be padded with zeros on the left
+ * @returns {String} - a string representation of the number with padding as needed
+ */
+publish.padZeros = function (num, width) {
+    width = width || 2;
+    var numstr = num ? num.toString() : "";
+
+    for (var i = numstr.length; i < width; i++) {
+        numstr = "0" + numstr;
+    }
+
+    return numstr;
+};
+
+/**
+ * Creates a date object from a git timestamp
+ *
+ * @param timestamp {String} - timestamp in seconds as returned by "git show -s --format=%ct HEAD"
+ * @returns {Object} - an object of date properties
+ */
+publish.fromTimestamp = function (timestamp) {
+    var timestampInMS = Number(timestamp) * 1000;
+    var date = new Date(timestampInMS);
+
+    return {
+        year: date.getUTCFullYear(),
+        month: date.getUTCMonth() + 1, // months are zero based by default
+        day: date.getUTCDate(),
+        hours: date.getUTCHours(),
+        minutes: date.getUTCMinutes(),
+        seconds: date.getUTCSeconds()
+    };
+};
+
+/**
+ * Converts a git timestamp into an ISO8601 timestamp
+ *
+ * @param timestamp {String} - timestamp in seconds as returned by "git show -s --format=%ct HEAD"
+ * @returns {String} - the time in the ISO8601 format yyyymmddThhmmssZ
+ */
+publish.convertToISO8601 = function (timestamp) {
+    var date = publish.fromTimestamp(timestamp);
+
+    for (var key in date) {
+        date[key] = publish.padZeros(date[key]);
+    }
+
+    return date.year + date.month + date.day + "T" + date.hours + date.minutes + date.seconds + "Z";
+};
+
+/**
+ * Throws an error if there are any uncommitted changes
+ *
+ * @param options {Object} - e.g. {"changes": "git status -s -uno"}
+ * @throws Error - An error object with a message containing a list of uncommitted changes.
+ */
+publish.checkChanges = function (options) {
+    var cmdStr = options.changes || defaults.changes;
+    var changes = publish.execSync(cmdStr);
+    if (changes.length) {
+        throw new Error("You have uncommitted changes\n" + changes);
+    }
+};
+
+/**
+ * Updates the package.json version to the specified version
+ * This does not commit the change, it will only modify the file.
+ *
+ * @param version {String} - the version to set in the package.json file
+ * @param options {Object} - e.g. {"version": "npm version --no-git-tag-version ${version}"}
```

Comment:
I think this options property is confusingly named - perhaps "versionCommand" or some such would be more descriptive